### PR TITLE
rv area management route

### LINKED ISSUE #43: create route for area-management page
this is the general desired structure of the output payload:
[ {
"areaId": number,
"areaName": name,
"active": true/false,
"siteList": [ { "siteId": num1, "siteName": name1}, { "siteId": num2, "siteName": name2}, .... ]
"countMT": number,
"countStudents": number,
"countSites": number
},
{
<repeat what's above for next area>
},
....
]

### PR DESCRIPTION
**Pull request by ctc-devops**
closes #43

### CODE REVIEW

File: routes/areas.js [hunk 1]
```diff
@@ -10,6 +10,45 @@ Do not think we need a get areas by year because can just
 use the get all areas request then filter based on year on frontend
 */
 
+// get an area by id
+router.get('/area-management', async (req, res) => {
+  try {
+    const areaInfo = await pool.query(`
+      SELECT area.*, site.num_sites, site.site_info, mt_info.num_mts, student_info.num_students
```
Comment (anikait-rao) [inline routes/areas.js:17]: Do you know if there's a way to also return the date the area was created? If so can you also include that as one of the things returned

Comment (anikait-rao) [inline routes/areas.js:17]: Also curious: num_sites, num_mts, and num_students are all returned as strings (it seems like one of the things that COUNT does). If you could return it as an INTEGER I think that may be a bit better for calculation

**Review by anikait-rao — CHANGES_REQUESTED** [review]
see comments

File: routes/areas.js (hunk 1 above)
Comment (Vakhshoori101) [reply routes/areas.js:17]: not possible in postgresql

Comment (Vakhshoori101) [reply routes/areas.js:17]: fixed

File: routes/areas.js [hunk 2]
```diff
@@ -10,6 +10,45 @@ Do not think we need a get areas by year because can just
...
+          (SELECT site.area_id, COUNT(site.site_id)::int as num_sites, array_agg(json_build_object('site_id', site.site_id, 'site_name', site.site_name)) as site_info
+          FROM site
+          GROUP BY site.area_id)
+        AS site ON site.area_id = area.area_id
+        LEFT JOIN
+          (SELECT site.area_id, COUNT(*)::int as num_mts
+          FROM master_teacher_site_relation as mt
+            INNER JOIN
+              (SELECT site.site_id, site.area_id FROM site)
+            AS site on site.site_id = mt.site_id
```
Comment (anikait-rao) [inline routes/areas.js:29]: sorry one more thing - it looks like the count of MTs is the number of rows in the MT/site relation where site id matches. However, teachers can be assigned to multiple sites. Can you count only distinct MTs when site id matches?

**Review by anikait-rao — CHANGES_REQUESTED** [review]
see comments

File: routes/areas.js (hunk 2 above)
Comment (anikait-rao) [reply routes/areas.js:29]: i don't believe this problem exists with counting the number of sites and students. students can only be assigned to one site, and all sites will be unique anyways.

Comment (Vakhshoori101) [reply routes/areas.js:29]: whoops good catch! should be fixed now

**Review by anikait-rao — APPROVED** [review]
looks good

**Pull request merged**